app/models/court_case.rb

```diff
@@ -0,0 +1,35 @@
+class CourtCase
+  include ActiveModel::Serialization
+
+  attr_reader :id, :case_seq, :begin_date, :case_type, :case_info_number, :case_status, :agency_id
+
+  def build_from_nomis(court_case)
+
+    @case_info_number = court_case['caseInfoNumber']
+    @id = @case_info_number
+
+    @case_seq = court_case['caseSeq']
+    @begin_date = court_case['beginDate']
+    @case_type = court_case['caseType']
+    @case_status = court_case['caseStatus']
+
+    @agency_id = court_case['agency']['agencyId']
+
+    self
+  end
+
+  def location_id
+    location = Location.find_by nomis_agency_id: @agency_id
+    if location
+      location.id
+    end
+  end
+
+  def location
+    location = Location.find_by nomis_agency_id: @agency_id
```

Comment:
Looks like an opportunity to DRY up the `Location` lookup - perhaps by amending this method to return the `Location` model rather than the `LocationSerializer`? I would have thought that it's not the models responsibility to perform JSON serialisation 🤔 

So this would then simply be

```ruby
def location
  Location.find_by nomis_agency_id: @agency_id
end
```

or, even better this gives an opportunity to cache the location lookup:

```ruby
def location
  @location ||= Location.find_by nomis_agency_id: @agency_id
end
```

This then means the method above can be amended to

```ruby
def location_id
  location&.id
end
```

(or potentially this is simple enough that you can just use `location&.id` in the calling code).

Reply:
I agree, this should be simplified, potentially we do not need the `location_id` or the explicit use of the serializer